### PR TITLE
Restore Qiskit.org analytics

### PR DESCRIPTION
**Pull request by delapuente**
The PR restores the analytics performed in the Polymer version of the website.

### CODE REVIEW

File: plugins/segment-analytics.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      successFlag: true
+    }
+
+    if (milestoneName) {
+      segmentEvent = { ...segmentEvent, milestoneName }
+    }
+
+    window.bluemixAnalytics.trackEvent('Custom Event', segmentEvent)
+  }
+}
```
Comment (y4izus) [inline plugins/segment-analytics.ts]: When a variable is used more than once, I think that is better to extract it to a new one (like you did with `params`). It's the case of `bluemixAnalytics`, `digitalData`, `location` and `pageInfo`

Comment (delapuente) [reply plugins/segment-analytics.ts]: Oh, I see. OK.

Comment (y4izus) [reply plugins/segment-analytics.ts]: Non-blocking
Suggestion:

```
function trackClickEvent(context: AnalyticsContext, params: ClickEventParams) {
  const { action, objectType, milestoneName } = params
  const { bluemixAnalytics, digitalData, location } = context

  if (bluemixAnalytics && digitalData && location) {
    const { href: url, pathname: path } = location
    const { page } = digitalData

    let segmentEvent: CustomEvent = {
      productTitle: page.pageInfo.productTitle,
      category: page.pageInfo.analytics.category,
      url,
      path,
      action: `${url} - Button Clicked: ${action}`,
      objectType,
      successFlag: true,
      ...milestoneName
    }

    bluemixAnalytics.trackEvent('Custom Event', segmentEvent)
  }
}
```

And also I think that it would good to test if `page`, `pageInfo` and `page.pageInfo.analytics` has value because if they are undefined, the app could crash

Comment (delapuente) [reply plugins/segment-analytics.ts]: You don't need to test `location` since it cannot be `null`, nor `''`, nor `undefined`.

I can test for the existence of `productTitle` and `category`, though, and raise an Exception, or leave a warning in the console.

Please notice that `milestoneName` is a `string` so the syntax `...milestoneName` does not make sense inside an object.

Comment (y4izus) [reply plugins/segment-analytics.ts]: sorry, I forget the second part for `milestone`, it should be `...milestone && {milestone}`. I know that it's a string, but by doing this, `milestone` is added to the object only when it has value.

For the rest, I think that with a warning in console to notify the dev is enough :)

**Review by y4izus — APPROVED** [review]
